Example/GiniVision/ComponentAPICoordinator.swift

```diff
@@ -0,0 +1,269 @@
+//
+//  ComponentAPICoordinator.swift
+//  GiniVision_Example
+//
+//  Created by Enrique del Pozo Gómez on 9/25/17.
+//  Copyright © 2017 Gini. All rights reserved.
+//
+
+import Foundation
+import GiniVision
+import Gini_iOS_SDK
+
+final class ComponentAPICoordinator {
+    
+    fileprivate var document:GiniVisionDocument?
+    fileprivate var navigationController: UINavigationController?
+    fileprivate var tabBarController: UITabBarController?
+    fileprivate var analysisScreen: ComponentAPIAnalysisViewController? {
+        return navigationController?.childViewControllers.flatMap { $0 as? ComponentAPIAnalysisViewController }.first
```

Comment:
I'm not convinced about that. Can't you hold a normal reference to the view controller. For instance, when you present it, you can keep a reference to it.

Reply:
Since it is a small collection of view controllers (4 at most) I found it a good approach (flatMap does not have a good performance with big collections).
But yes, it could be confusing and less readable.